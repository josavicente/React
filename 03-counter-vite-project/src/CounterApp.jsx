import { useState } from 'react';
import PropTypes from 'prop-types';


export const CounterApp = ( { value } ) => {


      const [ counter, setCounter] = useState(0);

      const handleAdd = () => {
            // setCounter (counter + 1);
            setCounter( (c) => c + 1)
      };

      const handleSubstract = () => {
            // setCounter (counter - 1);
            setCounter( (c) =>  c - 1 )
      };

      const handleReset = () => {
            // setCounter (0);
            setCounter( (c) => 0)
      };

      return ( 
      <>
            <h1>CounterApp</h1>
            <h2> { counter } </h2>
            {/* <button onClick = { (event) => handleEvent(event) }>
                  +1
            </button> */}
            <button onClick = { handleAdd }>
                  +1
            </button>
            <button onClick = { handleSubstract }>
                  -1
            </button>
            <button onClick = { handleReset }>
                  Reset
            </button>

      </>
)};


CounterApp.propTypes = {
      value: PropTypes.number
};

CounterApp.defaultProps = {
      value: 100
}