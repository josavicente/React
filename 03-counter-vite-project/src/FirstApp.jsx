
// const getResult = () => {
//       return 'Josa';
// }
import PropTypes from 'prop-types';

export const FirstApp = ( {title, message} ) => {

      return ( 
      <>
      <h1 data-testid="test-title">{title}</h1>
      <p>{message}</p>
      </>
)};

FirstApp.propTypes = {
      title: PropTypes.string.isRequired,
      message: PropTypes.string
};

FirstApp.defaultProps = {
      // title: 'No hay t√≠tulo',
      message: '666'
}
