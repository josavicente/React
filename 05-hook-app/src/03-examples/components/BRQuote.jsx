import { useState } from "react";
import { useLayoutEffect, useRef } from "react";


export const BRQuote = ({quote, author}) => {

    const pRef = useRef();
    const [boxSize, setBoxSize] = useState({ width: 0, height: 0 })
    
    useLayoutEffect(() => {
        const { height, width } = pRef.current.getBoundingClientRect();
        setBoxSize({width, height});
    }, [quote])

    return (
        <>
            <blockquote className="blockquote text-end" style={ { display: 'flex' } }>
                <p ref={ pRef } className="mb-1">{quote}</p>
                <footer className="blockquote-footer mt-2">{author}</footer>
            </blockquote>

            <code>
                {JSON.stringify(boxSize)}
            </code>
        </>
    );
};
