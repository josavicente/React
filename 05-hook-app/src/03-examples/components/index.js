export * from './BRQuote';
export * from './LoadingQuote';