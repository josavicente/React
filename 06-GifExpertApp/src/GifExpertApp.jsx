

//  const apiKey = '0SQ78YHAvJ2yMgLzOGGpwnMJ0N8cV95t';
//  const resp   = await fetch(`http://api.giphy.com/v1/gifs/random?api_key=${ apiKey }`);
//  const { data } = await resp.json(); 

export const GifExpertApp = () => {
      return (
            <>
            {/* TÃ­tulo */}
            <h1>GifExpertApp</h1>

            {/* Entrada de datos */}

            {/* Listado de Gif's */}
                  {/* GifItem */}
            </>
      )
};