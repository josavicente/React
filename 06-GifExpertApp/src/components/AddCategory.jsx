import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory }) => {

      const [inputValue, setInputValue] = useState('');
      
      const onInputChange = (event) => {
            // console.log(event.target.value);
            setInputValue(event.target.value);
      }

      const onSubmit = (event) => {
            event.preventDefault();
            const newInputValue = inputValue.trim();
            if (newInputValue.length <=1) return;
            
            //setCategories( categories => [ inputValue, ...categories ] );
            onNewCategory( newInputValue );
            setInputValue('');

      };

      return (
            <form onSubmit={ (event) => onSubmit(event) } aria-label="form">
                  <input
                  type="text"
                  placeholder="Buscar gifs"
                  value={ inputValue }
                  onChange={ onInputChange }
                  />
            </form>
            
      )
}

AddCategory.propTypes = {
      onNewCategory: PropTypes.func.isRequired,
} 
