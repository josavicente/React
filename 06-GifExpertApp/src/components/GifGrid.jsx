
import { useState, useEffect } from 'react';
import {getGifs} from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid =  ({category}) => {

      const [images, setImages] = useState([])

      const getImages = async() => {
            const newImages = await getGifs(category);
            console.log(newImages);
            setImages(newImages);
      }
      useEffect( () => {
            getImages();
            // getGifs(category)
            //       .then( newImages => setImages(newImages) );
      }, [] )

      return (
            <>
                  <h3>{category}</h3>
                  <div className="card-grid">
                  { images.map( (image) => (
                        <GifGridItem 
                              key={image.id}
                              {...image}
                              // title={image.title}
                              // image={image.url}
                              />
                        
                        )
                  )}
                  </div>
            </>
      )
}

