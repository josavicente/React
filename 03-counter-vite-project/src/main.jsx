import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
import { HelloWorldApp, } from './HelloWorldApp';
import './styles.css';


ReactDOM.createRoot(document.getElementById('root')).render(
      <React.StrictMode>
            <FirstApp  />
      </React.StrictMode>
);