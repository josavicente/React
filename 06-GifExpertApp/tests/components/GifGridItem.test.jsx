import { render, screen } from "@testing-library/react";
import { GifGridItem } from "../../src/components";

describe('Pruebas en GifGridItem', () => {

    const title = 'Goku';
    const url = 'http://jopsavicente.dev/';

    test('Debe de hacer match con el snapshot ', () => {  

        const { container } = render( <GifGridItem title={title} url={url}/> )
        expect( container ).toMatchSnapshot();

    });

    test('Debe de mostrar la imagen con el URL y ALT indicado', () => {  

        render( <GifGridItem title={title} url={url}/> )
        //screen.debug();
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('De de mostrar el tÃ­tulo en el component', () => { 


        render( <GifGridItem title={title} url={url}/> )
        expect(screen.getByText(title)).toBeTruthy();

    });


});
