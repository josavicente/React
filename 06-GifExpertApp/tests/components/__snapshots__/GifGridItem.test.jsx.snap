// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifGridItem Debe de hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Goku"
      src="http://jopsavicente.dev/"
    />
    <p>
      Goku
    </p>
  </div>
</div>
`;
